src/main/java/io/openliberty/tools/intellij/actions/RunLibertyDevTask.java

```diff
@@ -104,8 +107,10 @@ public void actionPerformed(@NotNull AnActionEvent e) {
             Component[] components = libertyWindow.getComponents();
 
             for (Component comp : components) {
-                if (comp.getName() != null && comp.getName().equals(Constants.LIBERTY_TREE)) {
-                    Tree libertyTree = (Tree) comp;
+                if (comp instanceof JBScrollPane && comp.getName() != null && comp.getName().equals(Constants.LIBERTY_SCROLL_PANE)) {
+                    JBScrollPane scrollPane = (JBScrollPane) comp;
+                    if (scrollPane.getViewport().getView() instanceof Tree) {
+                        Tree libertyTree = (Tree) scrollPane.getViewport().getView();
```

Comment:
The other opportunity for improvement for this code has to do with the fact that there is code duplication between the `update` and `actionPerformed` methods (once again, I know this code was already here and not something you wrote). We should pull out the duplicate code and put it in a separate method or methods that we can call from `update` and from `actionPerformed`.

The duplication is between this section: https://github.com/anusreelakshmi934/liberty-tools-intellij/blob/bugfix/%23175/src/main/java/io/openliberty/tools/intellij/actions/RunLibertyDevTask.java#L55-L79   
...and this section: https://github.com/anusreelakshmi934/liberty-tools-intellij/blob/bugfix/%23175/src/main/java/io/openliberty/tools/intellij/actions/RunLibertyDevTask.java#L101-L126

Please find the duplicate code and separate it into methods that can be called from `update` and `actionPerformed`.

Reply:
@anusreelakshmi934 the common method you created is very good. We can go ahead and keep it as is. I have just a few tips for improvement in the future. 

Typically, when creating common methods, you want to make it so they can be reused without having to modify them in the future. With the way that your common method is structured, if a new method is written that needs to handle the incoming "Action Event" on the Liberty Tree in a new way, your common method will need to be updated to handle the new scenario. This means the `boolean` parameter `isUpdate` won't be useful anymore (since there is now a third option) and new code will need to be added in the section after retrieving the `lastPathComponent`. 

It would be a better design to keep the unique parts of `update` and `actionPerformed` as part of those methods and set up the common method (or methods) so that it does not need to be modified for new methods to make use of it. Just something to keep in mind moving forward.